docs/source/frameworks/transformers.rst

```diff
@@ -1,38 +1,312 @@
+        "default": {},
+        "type": "text",
+    }
+    SUPPORTED_TASKS[TASK_NAME] = TASK_DEFINITION
+
+    classifier = pipeline(
+        task=TASK_NAME,
+        model=AutoModelForSequenceClassification.from_pretrained(
+            "distilbert-base-uncased-finetuned-sst-2-english"
+        ),
+        tokenizer=AutoTokenizer.from_pretrained(
+            "distilbert-base-uncased-finetuned-sst-2-english"
+        ),
+    )
+
+Once a new pipeline is added to the Transformers supported tasks, it can be saved to the BentoML model store with the additional 
+arguments of :code:`task_name` and :code:`task_definition`, the same arguments that were added to the Transformers :code:`SUPPORTED_TASKS` 
+when creating the pipeline. :code:`task_name` and :code:`task_definition` will be saved as model options alongside the model.
+
+.. code-block:: python
+   :caption: train.py
+    
+    import bentoml
+
+    bentoml.transformers.save_model(
+        "my_classification_model",
+        pipeline=classifier,
+        task_name=TASK_NAME,
+        task_definition=TASK_DEFINITION,
+    )
+
+Serving a Custom Pipeline
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To serve a custom pipeline, simply create a runner and service with the previously saved pipeline. :code:`task_name` and 
+:code:`task_definition` will be automatically applied when initializing the runner.
+
+.. code-block:: python
+    :caption: service.py
+    
+    import bentoml
+
+    from bentoml.io import Text, JSON
+
+    runner = bentoml.transformers.get("my_classification_model:latest").to_runner()
+
+    svc = bentoml.Service("my_classification_service", runners=[runner])
+
+    @svc.api(input=Text(), output=JSON())
+    def classify(input_series: str) -> list:
+        return runner.run(input_series)
+
+Adaptive Batching
+-----------------
+
+If the model supports batched interence, it is recommended to enable batching to take advantage of the adaptive batching capability 
+in BentoML by overriding the :code:`signatures` argument with the method name (:code:`__call__`), :code:`batchable`, and :code:`batch_dim` 
+configurations when saving the model to the model store . 
+
+.. seealso::
+
+   See :ref:`Adaptive Batching <guides/batching:Adaptive Batching>` to learn more.
+
+.. code-block:: python
+    :caption: train.py
+
+    import bentoml
 
-.. currentmodule:: bentoml.transformers
+    bentoml.transformers.save_model(
+        name="unmasker",
+        pipeline=unmasker,
+        signatures={
+            "__call__": {
+                "batchable": True,
+                "batch_dim": 0,
+            },
+        },
+    )
 
-.. autofunction:: bentoml.transformers.save_model
+Serving on GPU
+--------------
 
-.. autofunction:: bentoml.transformers.load_model
+BentoML Transformers framework will enable inference on GPU if the hardware is available.
 
-.. autofunction:: bentoml.transformers.get
+.. seealso::
 
+   See :ref:`Serving with GPU <guides/gpu:Serving with GPU>` to learn more.
```

Comment:
Lets comment this out for now. Since we don't have a docs on serving on GPU yet, I don't think this is useful to expose to the end user. We can add this back when our GPU docs are in place.